nixos/modules/virtualisation/xen-dom0.nix

```diff
@@ -1,213 +1,733 @@
+    (mkRemovedOptionModule
+      [
+        "virtualisation"
+        "xen"
+        "bridge"
+        "forwardDns"
+      ]
+      "The Xen Network Bridge options are currently unavailable. Please set up your own bridge manually."
+    )
+    (mkRenamedOptionModule
+      [
+        "virtualisation"
+        "xen"
+        "qemu-package"
+      ]
+      [
+        "virtualisation"
+        "xen"
+        "qemu"
+        "package"
+      ]
+    )
+    (mkRenamedOptionModule
+      [
+        "virtualisation"
+        "xen"
+        "package-qemu"
+      ]
+      [
+        "virtualisation"
+        "xen"
+        "qemu"
+        "package"
+      ]
+    )
+    (mkRenamedOptionModule
+      [
+        "virtualisation"
+        "xen"
+        "stored"
+      ]
+      [
+        "virtualisation"
+        "xen"
+        "store"
+        "path"
+      ]
+    )
+    (mkRenamedOptionModule
+      [
+        "virtualisation"
+        "xen"
+        "storePath"
+      ]
+      [
+        "virtualisation"
+        "xen"
+        "store"
+        "path"
+      ]
+    )
   ];
 
-  ###### interface
+  ## Interface ##
 
-  options = {
+  options.virtualisation.xen = {
 
-    virtualisation.xen.enable =
-      mkOption {
-        default = false;
-        type = types.bool;
-        description = ''
-            Setting this option enables the Xen hypervisor, a
-            virtualisation technology that allows multiple virtual
-            machines, known as *domains*, to run
-            concurrently on the physical machine.  NixOS runs as the
-            privileged *Domain 0*.  This option
-            requires a reboot to take effect.
-          '';
-      };
+    enable = lib.options.mkEnableOption "the Xen Hypervisor, a virtualisation technology defined as a *type-1 hypervisor*, which allows multiple virtual machines, known as *domains*, to run concurrently on the physical machine. NixOS runs as the privileged *Domain 0*. This option requires a reboot into a Xen kernel to take effect";
 
-    virtualisation.xen.package = mkOption {
-      type = types.package;
-      defaultText = literalExpression "pkgs.xen";
-      example = literalExpression "pkgs.xen-light";
-      description = ''
-        The package used for Xen binary.
-      '';
-      relatedPackages = [ "xen" "xen-light" ];
+    debug = lib.options.mkEnableOption "Xen debug features for Domain 0. This option enables some hidden debugging tests and features, and should not be used in production";
+
+    trace = lib.options.mkOption {
+      type = lib.types.bool;
+      default = cfg.debug;
+      defaultText = lib.options.literalExpression "false";
+      example = true;
+      description = "Whether to enable Xen debug tracing and logging for Domain 0.";
     };
 
-    virtualisation.xen.package-qemu = mkOption {
-      type = types.package;
-      defaultText = literalExpression "pkgs.xen";
-      example = literalExpression "pkgs.qemu_xen-light";
+    package = lib.options.mkOption {
```

Comment:
Will mkPackageOption fit?

Reply:
Same with qemu/stored

Reply:
Not if we want to keep `relatedPackages`.

Reply:
`error: function 'mkPackageOption' called with unexpected argument 'relatedPackages'`

Reply:
Thats... Unexpected for mkPackageOption.